src/corecel/cont/Bitset.hh

```diff
@@ -0,0 +1,310 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/cont/Bitset.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <climits>
+#include <cstddef>
+
+#include "corecel/Config.hh"
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/cont/detail/BitsetUtils.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Device-compatible bitset implementation.
+ *
+ * This implementation is based on libstdc++'s std::bitset implementation.
+ * It it a subset of the C++ standard, but it should be sufficient
+ * for our current use case. Given that GPU typically use 32-bit words, this
+ * uses unsigned int as the word type instead of the unsigned long used by the
+ * standard library.
+ *
+ * The following methods are not implemented:
+ * - conversions to string, to_ulong, to_ullong
+ * - set operations with other bitsets
+ * - stream operators
+ * - hash support
+ * - construct from string, from_ulong, from_ullong
+ */
+template<size_t N>
+class Bitset
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using word_type = unsigned int;
+    //!@}
+
+    static constexpr size_t bits_per_word = CHAR_BIT * sizeof(word_type);
```

Comment:
Since the original class was in `namespace std` this was fine but now we probably want either `std::size_t` or typedef a `size_type`.